lib/coinbase/webhook.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+module Coinbase
+  # A representation of a Webhook.
+  # This class provides methods to create, list, update, and delete webhooks
+  # that are used to receive notifications of specific events.
+  class Webhook
+    class << self
+      # Creates a new webhook for a specified network.
+      #
+      # @param network_id [String] The network ID for which the webhook is created.
+      # @param notification_uri [String] The URI where notifications should be sent.
+      # @param event_type [String] The type of event for the webhook (e.g., erc20_transfer).
+      # @param event_filters [Array<Hash>] Filters applied to the events that determine
+      #   which specific events trigger the webhook. Each filter should be a hash that
+      #   can include keys like `contract_address`, `from_address`, or `to_address`.
+      # @return [Coinbase::Webhook] A new instance of Webhook.
+      #
+      # @example Create a new webhook
+      #   webhook = Coinbase::Webhook.create(
+      #     network_id: 'ethereum',
+      #     notification_uri: 'https://example.com/callback',
+      #     event_type: 'transaction',
+      #     event_filters: [{ 'contract_address' => '0x...', 'from_address' => '0x...', 'to_address' => '0x...' }]
+      #   )
+      def create(network_id:, notification_uri:, event_type:, event_filters:)
```

Comment:
Are `event_filters` optional? If so they should have a default value of either `nil` or `[]`.

Is there a sane default `event_type` or do we want users to always have to specify something there?

Reply:
Also can I use this to monitor ETH transfer events as well or just ERC20 and ERC721?

Reply:
not yet, but there will be blockchain transactions (e.g. eth transfer events) coming out
event filters are not optional, user needs to specify at least an address